Jenkinsfile

```diff
@@ -0,0 +1,131 @@
+def registry = '562955126301.dkr.ecr.eu-west-2.amazonaws.com'
+def regCredsId = 'ecr:eu-west-2:ecr-user'
+def kubeCredsId = 'awskubeconfig002'
+def imageName = 'ffc-demo-api-gateway'
+def testImageName = 'ffc-demo-api-gateway-test'
+def repoName = 'ffc-demo-api-gateway'
+def branch = ''
+def pr = ''
+def mergedPrNo = ''
+def containerTag = ''
+
+def getMergedPrNo() {
+    def mergedPrNo = sh(returnStdout: true, script: "git log --pretty=oneline --abbrev-commit -1 | sed -n 's/.*(#\\([0-9]\\+\\)).*/\\1/p'").trim()
+    return mergedPrNo ?: ''
```

Comment:
should be `return mergedPrNo ? "pr$mergedPrNo" : ''` as the PR number returned is only the number, but the actual namespace that needs to be undeployed later has the "pr" prefix.

Reply:
Took me a while to track it down but I can see where that causes an issue. I don't like returning a non-numeric string from a function whose name indicates that a number should be returned (or indeed, setting a variable named `mergedPrNo` to a non-numeric value).

To me, it would feel more natural to add the `pr` prefix as required (in the undeploy stage):
```
undeployPR(kubeCredsId, imageName, "pr$mergedPrNo")
```

Would that be ok, or do you think it's better to just match the pipelines that have already been done?

Reply:
Looking at the number of pipelines already in review, I think it makes more sense to stick with the solution already in use for now. We'll be abstracting the Jenkinsfile functions to a library next sprint anyway, so that's a good opportunity to make improvements.

I've pushed the change you suggested.